flocx_ui/test/tests/test_flocx_rest_api.py

```diff
@@ -0,0 +1,19 @@
+from unittest import mock
+
+from openstack_dashboard.test import helpers as test
+
+from flocx_ui.api.flocx_rest_api import Offer as OfferAPI
+from flocx_ui.test.tests.helpers import get_test_data, RequestFactory
+
+class RestApiTests(test.TestCase):
+    @mock.patch('flocx_ui.api.flocx.offer_list')
+    def test_get_offers(self, mock_offer_list):
+        testData = get_test_data('offer_list_response')
+        mock_offer_list.return_value = testData
+
+        rf = RequestFactory()
+        request = rf.get('/api/flocx/offer/')
+
+        offerAPI = OfferAPI()
+        response = offerAPI.get(request)
+        self.assertEqual(response.json, testData)
```

Comment:
Add negative tests as well

Reply:
Handling of those sorts of errors is done automatically using the `@rest_utils.ajax()` decorator. You can see where that is implemented in the [`flocx_rest_api.py` file](https://github.com/jdtzmn/flocx-ui/blob/add-tests/flocx_ui/api/flocx_rest_api.py) or view the actual decorator source code [here](https://github.com/openstack/horizon/blob/master/openstack_dashboard/api/rest/utils.py#L76).

I do write negative tests where applicable (see the `test_create_invalid_offer` test in [this file](https://github.com/CCI-MOC/flocx-ui/pull/7/files#diff-060a18ea8795e0578bf89fa9153088e5R40) in PR #7), but I’m not sure that it makes sense here.

Let me know if this is not what you meant.

Reply:
Yes. That is what I was thinking